sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -1,129 +1,170 @@
+# `add` will throw a ValueError if added size results in the batch exceeding the maximum batch size.
+event_data_batch = producer.create_batch()
+event_data_batch.add(EventData('Single message'))
+producer.send_batch(event_data_batch)
+```
 
-In V1, you create an `EventHubClient`, then create a `AsyncSender`, and call `AsyncSender.send` to send an event that may have
-a list/generator of messages.
+### Receiving events 
 
-In V5, this has been consolidated into a one method - `EventHubProducerClient.send_batch`.
-Batching merges information from multiple events into a single send, reducing
-the amount of network communication needed vs sending events one at a time.
-This method deterministically tells you whether the batch of events are sent to the event hub.
+- The `run` and `stop` methods were previously used since the single `EventHubClient` controlled the lifecycle for all senders and receivers. In v5, the `run` and `stop` methods are deprecated since the `EventHubConsumerClient` controls its own lifecycle.
+- The `add_receiver` method is no longer used to create receiver clients. Instead, the `EventHubConsumerClient` is used for receiving events.
+- In v1, the `receive` method returned a list of `EventData`. You would call this method repeatedly every time you want receive a set of events. In v5, the new `receive` method takes user callback to process events and any resulting errors. This way, you call the method once and it continues to process incoming events until you stop it.
+- Additionally, we have a method `receive_batch` which behaves the same as `receive`, but calls the user callback with a batch of events instead of single events. 
+- The same methods can be used whether you want to receive from a single partition or from all partitions.
 
-So in V1:
+In v1:
 ```python
-client = EventHubClientAsync.from_connection_string(connection_str, eventhub=EVENTHUB_NAME)
-sender = client.add_async_sender(partition="0")
-try:
-    await client.run_async()
-    event_data = EventData(b"A single event")
-    await sender.send(event_data)
-finally:
-    await client.stop_async()
+client = EventHubClient(address)
+receiver = client.add_receiver(consumer_group, partition)
+client.run()
+batch = receiver.receive()
+client.stop()
 ```
 
-In V5:
+In v5:
 ```python
-producer = EventHubProducerClient.from_connection_string(conn_str=EVENT_HUB_CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
-async with producer:
-    event_data_batch = await producer.create_batch(partition_id="0")
-    event_data_batch.add(EventData(b"A single event"))
-    await producer.send_batch(event_data_batch)
+# Receive
+def on_event(partition_context, event):
+    print("Received event from partition: {}.".format(partition_context.partition_id))
+
+consumer_client = EventHubConsumerClient.from_connection_string(conn_str, consumer_group, eventhub_name=eh_name)
+with consumer_client:
+    consumer_client.receive(on_event=on_event)
```

Comment:
From a migration guide perspective, in v1 a receiver is only targeted at one partition,
however in v5 if no partition id is set, then consumer would receive events from all partition.

So I'm thinking for the sake of parity with v1 receiver case, in v5 we also receive events from a single partition.

```suggestion
    consumer_client.receive(on_event=on_event, partition_id=partition_id)
```